tests/test_edge_cases.py

```diff
@@ -1306,10 +1377,12 @@ class Model(BaseModel):
         h: Union[Type[FooBar], Sequence[Type[FooBar]]] = FooBar
         i: Union[Type[FooBar], Sequence[Type[FooBar]]] = [FooBar]
 
-    assert Model.model_fields.keys() == {'b', 'c', 'e', 'f', 'g', 'h', 'i'}
+        model_config = dict(arbitrary_types_allowed=True)
+
+    # TODO: Why was this ever not the case? I don't understand why these weren't all fields
```

Comment:
because the value of `a` is `int` which meant it got omitted from fields.

But requiring typing annotations on all fields, we simplified the extraction of fields from `namepace` / `attributes` a lot.